### ISSUE TITLE
Why it exist negative number after computed finally?

### ISSUE THREAD

**Issue by xhappy** [qijiezhao/pseudo-3d-pytorch#4]
Hello QiJie,

Thanks for your great work.

I am doing experiment to confirm whether your pre-trained models works for my requests, I have made the codes working well, but I got many negative numbers and the numbers (more than 1, such as 2.8574)from the finally predicted results as below. Could you please help me? Thanks in advance. 

Columns 0 to 9
 0.2547  **2.8574**  0.2466  0.5315  0.3087  0.8649  0.4103 **-0.6601** -1.4141 -0.2270
Columns 10 to 19 
-0.7751  0.1545  0.3906  0.5003  0.7063 -0.7299  1.2138 -0.9078 -0.5577 -0.3110
Columns 20 to 29 
 0.5861  0.3854 -2.2098 -1.2202 -0.6508 -0.7924  1.5414  0.1893 -0.7203 -0.6547
Columns 30 to 39 
-0.6998 -0.2219 -0.6565 -1.1104  0.5413  0.1847  1.1829  1.1412 -0.7261 -0.9064
Columns 40 to 49 
-0.3360  0.2168 -0.4412  0.0375  1.2200  0.1787  0.0969  0.3945  0.7286  1.1807
Columns 50 to 59 
 0.3262  0.3818 -0.3003  0.4018 -0.6653 -0.0746  0.2734  0.5684 -0.8847 -1.1838
Columns 60 to 69 
 1.1265 -0.1349 -0.2033  1.7768 -0.5521  1.4568 -0.4313  0.4930  0.0090  0.5893
Columns 70 to 79 
 0.4483  0.4776 -0.8270 -0.1999  0.4683 -0.6691 -0.1553  0.8590 -1.0097  1.4962
Columns 80 to 89 
-1.2980  1.0693 -0.1839  0.1081  0.3478 -0.6089  1.1642  0.4668 -0.8454 -0.2026
Columns 90 to 99 
-0.3866  0.6512 -0.9677 -1.0000 -0.5366 -0.9803  0.5690 -0.1478 -1.4045  0.0646
Columns 100 to 109
 1.9957  1.3143  0.3878  1.0429  0.5111  1.1285 -0.7205  0.5839 -0.1410  1.8795
Columns 110 to 119 
 0.6499  1.0486  2.6186  2.2763  1.4925  1.8187  1.1356  0.7856 -0.3900 -1.0579
Columns 120 to 129 
-0.6244 -0.0601  1.3689 -0.1556 -0.4125  0.4500 -0.4232  1.8318  1.0345  0.3353
Columns 130 to 139 
-0.1830 -0.3989 -0.4298 -0.1682 -1.2609 -0.7833 -0.3433  0.6277  0.5956 -0.6453
Columns 140 to 149 
-0.0364 -0.9813 -0.7421  0.1030 -0.6083  1.0187  0.0918 -0.3473 -0.4790  1.7018
Columns 150 to 159 
 2.5567 -0.4137  1.1153  0.1984 -0.5811  0.5228  0.2125  0.2379  1.6737  0.8861
Columns 160 to 169 
-1.0300 -0.3617 -1.4262 -1.1236  0.0033  0.4325 -1.3333 -1.7200  0.5755  0.2818
Columns 170 to 179 
-0.8611  0.1419 -0.5707 -0.8100 -1.3173  0.8708  1.1916 -1.5630 -0.1792 -0.6229
Columns 180 to 189 
 2.1504 -1.2198 -1.1930 -0.3797 -0.0317  0.1053  0.2186 -0.2541  0.7459 -1.1295
Columns 190 to 199 
-0.5629 -0.9077 -0.0785  0.2610 -0.0439 -0.9229  0.8159  0.9987  0.4082 -0.6426
Columns 200 to 209 
 0.5981  0.1352 -0.2190  1.8740  0.3333 -1.6436 -1.4595 -0.2055 -2.0410  0.0428
Columns 210 to 219 
 0.4188  0.4601  1.2329  0.6426 -0.0635 -0.0622 -0.1042  0.2755 -0.4148  0.7330
Columns 220 to 229 
 0.7128 -0.8897 -0.8962 -0.1775  0.1521 -0.4070 -1.3691  0.0299  0.7416  0.6440
Columns 230 to 239 
-0.1439 -0.1009  0.1973  1.7817 -0.6647 -0.8101  0.0690 -0.7127 -0.2753 -0.7935
Columns 240 to 249 
-0.9452 -0.0944 -0.2082  1.4317  0.5112 -0.3383 -0.7704  0.5039  1.2479  1.4883
Columns 250 to 259 
 0.9024 -0.3343  0.0966 -0.1657 -2.1660 -0.1980  2.3596 -0.5610  0.6940  0.2634
Columns 260 to 269 
-0.4295  1.0217  0.7458  0.5033  0.9743  0.4465 -0.5388 -0.0811 -0.7372  0.3217
Columns 270 to 279 
-0.7245 -1.0703 -1.4170 -0.3830 -0.4575  1.1041  0.9542  1.1681 -0.0909  1.3096
Columns 280 to 289 
-0.0418  0.1292 -1.1363  0.0784 -0.2900 -0.2781 -1.2245 -0.1769  2.4483  1.4832
Columns 290 to 299 
-0.5490  0.5404  1.6246  0.1691 -0.5037 -0.1408 -0.3451 -0.6183 -0.5691 -0.6429
Columns 300 to 309 
 0.3554  0.1019  0.7276 -0.1883  1.6244  0.5651 -0.0830 -1.6845 -1.5468 -1.4152
Columns 310 to 319 
-2.4527 -0.6569 -1.5320  0.2806  1.3925 -1.1757  1.0803  0.6409 -1.1188  1.2455
Columns 320 to 329 
 1.5661 -1.6605 -1.5542 -2.0289 -2.0886  0.9124 -0.5289  0.2374  0.2726 -1.4111
Columns 330 to 339 
-0.0333  1.6842 -0.3154  0.0249  0.5457 -1.2823 -1.0470 -0.6791  0.8771 -1.0866
Columns 340 to 349 
-1.7285 -1.8984 -1.0631  0.3520  0.5620  0.4164 -0.1236  0.2988 -0.2146  0.0687
Columns 350 to 359 
-0.7481 -1.0270  0.0341  0.7840 -0.7411  0.6185  0.2475  1.6499 -0.6291  0.8590
Columns 360 to 369 
-1.6230  1.5294 -0.3375  0.8977 -0.8415  0.4278 -0.3155 -1.3621 -0.3663  0.6557
Columns 370 to 379 
-0.1923 -0.5912 -0.3559 -0.3140  0.2376 -1.0435 -1.4257 -0.1515  1.7692  0.7018
Columns 380 to 389 
-0.2411  0.4794  1.0348 -1.2981 -0.9805  0.1539 -0.4159 -0.3165  0.5872  1.0687
Columns 390 to 399 
-0.2811 -0.5407  0.2960 -1.5490 -0.0983  0.1577 -0.6394  1.0029  0.1816 -0.9346
[torch.cuda.FloatTensor of size 1x400 (GPU 0)]

**Comment by qijiezhao** [comment]
Try to add a softmax layer. softmax layer has no parameters, so it is not included in the main framework.

**Comment by xhappy** [comment]
Thanks, The probability computed now is normal (at least its value range is between 0 and 1.) after add softmax layer.
